src/features/embalm/stepContent/hooks/useCreateSarcohpagus.ts

```diff
@@ -31,103 +52,183 @@ export function useCreateSarcophagus() {
     outerPublicKey,
     outerPrivateKey,
     selectedArchaeologists,
-    payloadTxId,
+    publicKeysReady,
     shardsTxId,
-    requiredArchaeologists,
+    requiredArchaeologists
   } = useSelector(x => x.embalmState);
   const { isUploading } = useSelector(x => x.bundlrState);
   // const { uploadFile } = useBundlr();
   const { uploadArweaveFile } = useArweaveService();
-  const { submitSarcophagus } = useSubmitSarcophagus();
-
-  // TODO: Move this into its own hook and check all fields
-  // TODO: render a message to the user about what data is missing
-  const canCreateSarcophagus = recipientState.publicKey !== '' && !!outerPublicKey && !!file;
-
-  const uploadAndSetDoubleEncryptedFile = useCallback(async () => {
-    const emptyData = { sarcophagusPayloadTxId: '' };
-
-    if (!canCreateSarcophagus) return emptyData;
-
-    const payload = await readFileDataAsBase64(file);
+  const { dialSelectedArchaeologists, initiateSarcophagusNegotiation } = useSarcophagusNegotiation();
+  const [currentStep, setCurrentStep] = useState(CreateSarcophagusStep.NOT_STARTED);
+  const [stepExecuting, setStepExecuting] = useState(false);
+
+  // Create Sarcophagus State
+  const [archaeologistShards, setArchaeologistShards] = useState([] as ArchaeologistEncryptedShard[]);
+  const [encryptedShardsTxId, setEncryptedShardsTxId] = useState('');
+  const [sarcophagusPayloadTxId, setSarcophagusPayloadTxId] = useState('');
+  const [archaeologistSignatures, setArchaeologistSignatures] = useState(new Map<string, string>([]));
+  const [negotiationTimestamp, setNegotiationTimestamp] = useState(0);
+
+  const arweaveTxIds = [sarcophagusPayloadTxId, encryptedShardsTxId];
```

Comment:
A few options for this:
1. We leave as-is
2. We setup a reducer for all these state values, still keep it local to this hook
3. Setup a new context/provider for this state that `useSubmitSarcophagus` could use
4. Revert to keeping these values in global state

I think there's some advantage to having the state variables generated during the createSarcophagus process not be global, as the user is going to be required to stay on the createSarcophagus page while this process is happening, and I don't think we want these values in global state were they to switch pages.

Reply:
1. This is a good option. I think it's completely acceptable to use state outside of the store, actually I would encourage it in this case. We should minimize the use of the store as much as we can (although we haven't been doing a great job of that).
2. I don't think a reducer is necessary in this case. Reducers are good when the next state depends on the previous state, or inside a context 
3. Setting up a dedicated context/provider for this is an option. If it makes sense to split up the create sarcophagus functions across multiple hooks/components then setting up a context/provider would be beneficial. It might bean over-engineered solution for this case though.
4. Any time we can remove something from the global store without making the application much more complicated will simplify the app overall.

Reply:
@knoll3 -- thanks for these comments,  this will help make decisions going forward --- will leave this code as-is for now.